sklearn/compose/_target.py

```diff
@@ -162,7 +162,7 @@ def fit(self, X, y, sample_weight=None):
         -------
         self : object
         """
-        y = check_array(y, accept_sparse=False, force_all_finite=True,
+        y = check_array(y, accept_sparse=False, force_all_finite=False,
```

Comment:
It should not be `'allow-nan'` instead?

Reply:
@glemaitre all checks are turned off as suggested by @jnothman [here](https://github.com/scikit-learn/scikit-learn/pull/11349#pullrequestreview-131437358)

Reply:
OK fine with me then.